torch/lib/c10d/ProcessGroupGloo.hpp

```diff
@@ -251,6 +289,10 @@ class ProcessGroupGloo : public ProcessGroup {
   c10::intrusive_ptr<ProcessGroup::Work> barrier(
       const BarrierOptions& opts = BarrierOptions()) override;
 
+  GlooStore* _getStore() {
```

Comment:
Can you change this to return a c10::intrusive_ptr? Generally we would like to avoid handling raw pointers and ensure safety with regards to things like ref counting, lifetimes, etc.

Reply:
I'm having trouble figuring out how to cast it 

I tried this and a couple of similar variants with `get()` without `get()`, casting the raw pointer, casting the object directly etc, dereferencing the store and all give the exact same issue

```
  c10::intrusive_ptr<GlooStore> _getStore() {
      auto store = c10::make_intrusive<GlooStore>(store_.get()); //also tried GlooStore*
      return store;
  }
```

./c10/util/intrusive_ptr.h:238:88: error: ‘class c10d::ProcessGroupGloo::GlooStore’ has no member named ‘refcount_’
     if (target_ != NullType::singleton() && detail::atomic_refcount_decrement(target_->refcount_) == 0) {

Same issue for `weakcount`